### PR TITLE
Fix `takeEvery` in `cluster` branch

### PR DESCRIPTION
**Pull request by jwoudenberg**
This fixes an issue with key identification in variadic commands.

We noticed a problem using the `mset` command. It threw a `CrossSlotException`, but inspection showed the keys it was trying to set definitely all fell in the same hashslot. Upon looking closer we noticed that the values rather than the keys were used to determine whether the command was for a single hash slot.

### CODE REVIEW

File: src/Database/Redis/Cluster/Command.hs [hunk 1]
```diff
@@ -121,9 +121,8 @@ readNumKeys (rawNumKeys:rest) = do
 readNumKeys _ = Nothing
```
Comment (michaelglass) [inline src/Database/Redis/Cluster/Command.hs:122]:
```suggestion
-- takeEvery 1 [1,2,3,4,5] ->[1,2,3,4,5]
-- takeEvery 2 [1,2,3,4,5] ->[1,3,5]
-- takeEvery 3 [1,2,3,4,5] ->[1,4]
```

Comment (jwoudenberg) [reply src/Database/Redis/Cluster/Command.hs]: Thank you!

**Comment by aravindgopall** [comment]
@jwoudenberg Thanks for raising the PR. If possible, can you fix the conflicts that would be helpful to merge this PR.

**Comment by jwoudenberg** [comment]
Hey @aravindgopall! I've moved to a different stack and haven't worked on this code for a while, so it'd be some effort for me to rebuild context and test the fixed version works as expected. Maybe @stoeffel has an interest in adopting this PR?

**Pull request merged**